src/boot/efi/meson.build

```diff
@@ -306,7 +323,7 @@ foreach tuple : [['systemd_boot.so', systemd_boot_efi_name, systemd_boot_objects
                 tuple[0],
                 input : tuple[2],
                 output : tuple[0],
-                command : [efi_ld, '-o', '@OUTPUT@', efi_ldflags, tuple[2], '-lefi', '-lgnuefi', libgcc_file_name],
+                command : [efi_cc, '-o', '@OUTPUT@', efi_ldflags, tuple[2], '-lefi', '-lgnuefi', '-lgcc'],
```

Comment:
what's the -lgcc thing? can you explain? shouldn't this be a separate commit?

Reply:
Basically, the compiler already knows the correct libgcc to link to, so we don't have to do it manually anymore. The `-lgcc` can probably be dropped completely now that I think of it...

Reply:
Nvm, explicit `-lgcc` is stil needed.